src/core/events/bpf.rs

```diff
@@ -580,6 +544,24 @@ pub(crate) struct BpfRawSectionHeader {
 
 unsafe impl Plain for BpfRawSectionHeader {}
 
+/// EventSection factory, providing helpers to create event sections from
+/// ebpf.
+///
+/// Please use `#[retis_derive::event_section_factory(SectionType)]` to
+/// implement the common traits.
+pub(crate) trait EventSectionFactory: RawEventSectionFactory {
```

Comment:
This is a bit confusing as it's now only required for the `RawEventSectionFactory` but still looks like a generic trait. Due to this we're also using `EventSectionFactory` all the place but those are only raw event factories. Not sure what is possible here, but would be cleaner to be able to use `RawEventSectionFactory` directly.

`Module::section_factory` should be renamed too at some point.

Reply:
Why not the opposite? There is only one factory, why not just call it EventSectionFactory? We could argue that we're not really "creating" an event from json, we're just unmarshaling it.

Reply:
Yes that would work too, my point was more that we have a layer of abstraction where we only have a single possible implementation.